internal/proto/server_adaptor.go

```diff
@@ -16,5 +17,45 @@ func NewGRPCServer(del tiltd.TiltD) *GRPCServer {
 var _ DaemonServer = &GRPCServer{}
 
 func (s *GRPCServer) CreateService(ctx context.Context, service *Service) (*CreateServiceReply, error) {
-	return &CreateServiceReply{}, s.del.CreateService(ctx, service.K8SYaml)
+	return &CreateServiceReply{}, s.del.CreateService(ctx, service.K8SYaml, service.DockerfileText, mountsP2D(service.Mounts), cmdsP2D(service.Steps), service.DockerfileTag)
+}
+
+func mountsP2D(mounts []*Mount) []build.Mount {
+	r := []build.Mount{}
+
+	for _, m := range mounts {
+		r = append(r, mountP2D(m))
+	}
+
+	return r
+}
+
+func mountP2D(mount *Mount) build.Mount {
+	return build.Mount{
+		// TODO(dmiller): convert to repo, we need a path
+		Repo:          build.LocalGithubRepo{LocalPath: ""},
```

Comment:
We need to modify the proto  so that we can get passed the path of the repo with the various kinds of repos. I'm regretting making this generic because dealing with generics in go is so painful.

Reply:
sigh